security/impersonating_user.rst

```diff
@@ -187,6 +187,179 @@ setting:
+                return false;
+            }
+
+            if (in_array('ROLE_CUSTOMER', $subject->getRoles()) 
+                && $this->hasSwitchToCustomerRole($token)) {
+                return true;
+            }
+
+            return false;
+        }
+
+        private function hasSwitchToCustomerRole(TokenInterface $token)
+        {
+            $roles = $this->roleHierarchy->getReachableRoles($token->getRoles());
+            foreach ($roles as $role) {
+                if ($role->getRole() === 'ROLE_SWITCH_TO_CUSTOMER') {
+                    return true;
+                }
+            }
+            
+            return false;
+        }
+    }
+
+.. caution::
+
+    Notice that when checking for the ``ROLE_CUSTOMER`` role on the target user, only the roles 
+    explicitly assigned to the user are checked rather than checking all reachable roles from
+    the role hierarchy.  The reason for this is to avoid accidentally granting access to an 
+    elevated user that may have inherited the role via the hierarchy.  This logic is specific 
+    to the example, but keep this in mind when writing your own voter.
+
+Next, add the roles to the security configuration:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+        
+        # config/packages/security.yaml
+        security:
+            # ...
+
+            role_hierarchy:   
+                ROLE_CUSTOMER:    [ROLE_USER]
+                ROLE_EMPLOYEE:    [ROLE_USER, ROLE_SWITCH_TO_CUSTOMER]
+                ROLE_SUPER_ADMIN: [ROLE_EMPLOYEE, ROLE_ALLOWED_TO_SWITCH]
+
+    .. code-block:: xml
+
+        <!-- config/packages/security.xml -->
+        <?xml version="1.0" encoding="UTF-8"?>
+        <srv:container xmlns="http://symfony.com/schema/dic/security"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:srv="http://symfony.com/schema/dic/services"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+            <config>
+                <!-- ... -->
+
+                <role id="ROLE_CUSTOMER">ROLE_USER</role>
+                <role id="ROLE_EMPLOYEE">ROLE_USER, ROLE_SWITCH_TO_CUSTOMER</role>
+                <role id="ROLE_SUPER_ADMIN">ROLE_EMPLOYEE, ROLE_ALLOWED_TO_SWITCH</role>
+            </config>
+        </srv:container>
+
+    .. code-block:: php
+
+        // config/packages/security.php
+        $container->loadFromExtension('security', array(
+            // ...
+
+            'role_hierarchy' => array(
+                'ROLE_CUSTOMER'    => 'ROLE_USER',
+                'ROLE_EMPLOYEE'    => 'ROLE_USER, ROLE_SWITCH_TO_CUSTOMER',
+                'ROLE_SUPER_ADMIN' => array(
+                    'ROLE_EMPLOYEE',
+                    'ROLE_ALLOWED_TO_SWITCH',
+                ),
+            ),
+        ));
+
+Thanks to autowiring, we only need to configure the role hierarchy argument when registering 
```

Comment:
`we only need to configure`. Correct me if I'm wrong, but I believe the Symfony docs point towards "you", rather than "we".